sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -752,6 +752,55 @@ class DataFrame private[sql](
   }
 
   /**
+   * Compute numerical statistics for given columns of this [[DataFrame]]:
+   * count, mean (avg), stddev (standard deviation), min, max.
+   * Each row of the resulting [[DataFrame]] contains column with statistic name
+   * and columns with statistic results for each given column.
+   * If no columns are given then computes for all numerical columns.
+   *
+   * {{{
+   *   df.describe("age", "height")
+   *
+   *   // summary age   height
+   *   // count   10.0  10.0
+   *   // mean    53.3  178.05
+   *   // stddev  11.6  15.7
+   *   // min     18.0  163.0
+   *   // max     92.0  192.0
+   * }}}
+   */
+  @scala.annotation.varargs
+  def describe(cols: String*): DataFrame = {
+
+    def aggCol(name: String = "") = s"'$name' as summary"
+    val statistics = List[(String, Expression => Expression)](
+      "count"  -> (expr => Count(expr)),
+      "mean"   -> (expr => Average(expr)),
+      "stddev" -> (expr => Sqrt(Subtract(Average(Multiply(expr, expr)),
```

Comment:
hm - i'm surprised we don't have a stddev expression yet. 

@marmbrus we can either create a stddev expression, or in the optimizer rewrite it to what's available here. What do you think?